Code/Framework/AzToolsFramework/Tests/Prefab/PrefabUndoAddEntityTestFixture.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Prefab/PrefabTestFixture.h>
+
+#include <AzToolsFramework/Prefab/Undo/PrefabUndoAddEntity.h>
+#include <AzToolsFramework/Prefab/Undo/PrefabUndoAddEntityAsOverride.h>
+
+namespace UnitTest
+{
+    using namespace AzToolsFramework::Prefab;
+    using namespace PrefabTestUtils;
+
+    using CreatePrefabUndoAddEntityNodeResult =
+        AZ::Outcome<PrefabUndoAddEntity, AZStd::string>;
+    using CreatePrefabUndoAddEntityAsOverrideNodeResult =
+        AZ::Outcome<PrefabUndoAddEntityAsOverride, AZStd::string>;
+    using InstanceList = AZStd::vector<AZStd::reference_wrapper<Instance>>;
+
+    class PrefabUndoAddEntityTestFixture
+        : public PrefabTestFixture
+    {
+    protected:
+        using PrefabTestFixture::CreateEntity;
+
+        EntityAlias CreateEntity(const AZStd::string& entityName,
+            Instance& owningInstance);
+
+        EntityAlias CreateEntity(const AZStd::string& entityName,
+            Instance& owningInstance, const EntityAlias& parentEntityAlias);
+
+        CreatePrefabUndoAddEntityNodeResult CreatePrefabUndoAddEntityNode(
+            const EntityAlias& newEntityAlias,
+            Instance& focusedInstance,
+            const AZStd::string& undoAddEntityOperationName = "Undo Adding Entity");
+
+        CreatePrefabUndoAddEntityNodeResult CreatePrefabUndoAddEntityNode(
+            const EntityAlias& parentEntityAlias, const EntityAlias& newEntityAlias,
+            Instance& focusedInstance,
+            const AZStd::string& undoAddEntityOperationName = "Undo Adding Entity");
+
+        CreatePrefabUndoAddEntityAsOverrideNodeResult CreatePrefabUndoAddEntityAsOverrideNode(
```

Comment:
Same here as well. All 3 flavors of CreatePrefabUndoAddEntityAsOverrideNode can be combined into 1